strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/StoragesConfigurationController.java

```diff
@@ -252,6 +258,13 @@ public ResponseEntity addOrUpdateRepository(@ApiParam(value = "The storageId", r
 
             configurationManagementService.saveRepository(storageId, repository);
 
+            if (repository != null && RepositoryTypeEnum.PROXY.getType().equals(repository.getType()))
+            {
+                proxyRepositoryConnectionPoolConfigurationService.setMaxPerRepository(
```

Comment:
It would be better to do it inside `org.carlspring.strongbox.services.impl.ConfigurationManagementServiceImpl#saveRepository` inside `modifyInLock` block